src/yb/master/master-path-handlers.cc

```diff
@@ -180,10 +180,10 @@ void MasterPathHandlers::CallIfLeaderOrPrintRedirect(
       }
 
       if (master.role() == consensus::RaftPeerPB::LEADER) {
-        // URI already starts with a /, so none is needed between $1 and $2.
-        redirect = Substitute("http://$0:$1$2$3",
-                              master.registration().http_addresses(0).host(),
-                              master.registration().http_addresses(0).port(),
+        // URI already starts with a /, so none is needed between $0 and $1.
+        redirect = Substitute("http://$0$1$2",
+                              HostPortToString(master.registration().http_addresses(0).host(),
```

Comment:
It worth to add HostPortToString version that accepts `HostPortPB`

Reply:
Yeah, that's a possibility.